### DISCUSSION TITLE
nginx config help

### DISCUSSION THREAD

**Discussion by RatedR9** [dani-garcia/vaultwarden#4161] [Q&A]
I recently upgraded to the latest version and used the nginx config provided by black but no login with device choice in the web, how do i do?

**Comment by RealOrangeOne** [comment]
Have you tried the proxy configurations from https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples? This is the best list. Unfortunately, just getting a 404 doesn't tell us much without more context as to your configuration.

**Reply by RatedR9** [reply]
yes i tried the config , i use blackdex config.nginx says 2023/12/12 13:29:09 [error] 592118#592118: *779 open() "/usr/share/nginx/html/vault/identity/connect/token" failed (2: No such file or directory), client: , server: , request: "POST /vault/identity/connect/token HTTP/2.0", host: ""

**Reply by RealOrangeOne** [reply]
Looks like the config is always trying to find the paths in the filesystem, rather than actually sending to vaultwarden itself.

You keep saying @BlackDex sent you a config, but without the actual config or reference, it's impossible to really help you.

**Reply by RatedR9** [reply]
now, i check the config and modified one mistake. but new problem come:no login with device choice in the web
upstream vaultwarden-default {
  zone vaultwarden-default 64k;
  server 127.0.0.1:8080;
  keepalive 2;
}
map $http_upgrade $connection_upgrade {
    default upgrade;
    ''      "";
}
server {
    
    listen 443 ssl http2;
    server_name    ;
 listen [::]:443 ssl http2;
 client_max_body_size 525M;
    ssl_protocols TLSv1.2 TLSv1.3;
location /vault/ {
proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection $connection_upgrade;

      proxy_set_header Host $host;
      proxy_set_header X-Real-IP $remote_addr;
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header X-Forwarded-Proto $scheme;
proxy_pass http://vaultwarden-default;
}
}

**Comment by RatedR9** [comment]
browser extension has the choice, and it worked but i don't receive notifications from my iphone.I manually approve it. finally i find when i unlocked the phone and unlocked vault ,pull down to refresh, i can see the approve window immediately.But does it automatically notify immediately?
